addons/knowledge/models/knowledge_article.py

```diff
@@ -222,6 +227,10 @@ def _compute_user_can_write(self):
                 article.user_can_write = member_permissions[article.id] == "write" if article.id in member_permissions \
                     else article_permissions[article.id] == 'write'
 
+    def _compute_user_can_invite(self):
+        for article in self:
+            article.user_can_invite = article.user_can_write or article.env.user.has_group('base.group_system')
```

Comment:
I removed the field as you asked but I think that it would be better to keep it because it would be a PITA for the community to change the invitation permissions without it: They would have to overwrite each backend functions and all templates doing those checks. 

With the computed field, they can simply overwrite the `_compute_user_can_invite` and put the condition they want and that's about it. It would also benefit us: if we decide to change the default invitation permissions, we will only have to update the compute method.

I also think that it helps clarifying the code a lot: Someone who is not familiar with the Knowledge code base may not know when a user can invite a new user. The condition is currently implicit and bluried in the code. Having a computed field indicating when a user is allowed to invite a new user would make it way more clearer.

When I started working on the CRM module, it was not really clear for me when a lead is considered as "lost": Overtime, the conditions that makes a lead lost changes and they were some inconsistencies in the code. I think that using a computed field to encapsulate part of the business logic is good practice.